### DISCUSSION TITLE
home assistant user

### DISCUSSION THREAD

**Discussion by chirs-51** [johanmeijer/grott#257] [Q&A]
i installed grott on my home assistant sever and its runs fine i can simply set det sever to responde to shine and that works but if i set it to lokal ip i can get the welcome screen and get data but i can't get its to chance value

ggmode: proxy
gnomqtt: false
gmqttip: 192.xxx.x.xxx
gmqttport: "1883"
gmqttauth: true
gmqtttopic: energy/grott
gmqttuser: my user name
gmqttpassword: my passwode
ginvtype: sph
ggrowattip: 47.91.67.66 < Works i get data from mqtt and shine (127.0.0.1 lokal ip mqtt still works and i can pull some recuest
gverbose: true
ggrowattport: 5279 online.  5781 lokal port i can se the welcome screen and get values
gincludeall: true
gsendbuf: false

i can see my ind log when starting its olso runs on port 5279 but but that port seems to have even more strange behavior

http://192.168.x.xxx:5782/datalogger?command=register&register=4&datalogger=JPCxxxxxxx
gets {"value": "5"} 8/10 times 

http://192.xxx.x.xxx:5782/inverter?command=register&register=1091&inverter=OMxxxxxxxx
gets anwser 8/10 times 

http://192.xxx.x.xxx:5782/datalogger?command=register&register=4&datalogger=JPCxxxxxxx&value=0.1
sometimes get  {"value": "5"}  and sometimes get  invalid datalogger id and most of the times nothing just a thinking chrome browser

http://192.xxxx.xxx5782/inverter?command=register&register=1091&inverter=OMxxxxxxxx&value=50

what i undersand of the Danish guide i got i needed to use the port 5781 to send commands but i never get eny feedback from that port. 

i thing this lokal sever is all i ever wish fore but its simes to a little tricky to setup and run. i wish the projekt the bedst!!! 

i hope someone can help me and even help somebody else with this post :)

**Comment by johanmeijer** [comment]
Do you use the grottserver docker image? 

Then you do not have to specify ggrowattip: 47.91.67.66  and ggrowattport: 5279. That will bypass the grottserver and send to the growtt server again. 

You need to open port 5279 (your inverter will send data to this port) and port 5782 (that is used for sending the commands).  
 Fro docker run: -p 5279:5279 -p 5782:5282.
 
 Be aware using grottserver will disable sending data to the growatt server, but gives you the ability to give API commands.

**Reply by chirs-51** [reply]
thangs for the answer and building a alsome solution for non china server bs!!! 

i will when its it working not send the data to shine but thanks for the info!

i'm running 2.7.6.1 i have tired do use som of the home assistant frogs but its semes there is the libscrc needs to be 1.8.1

i just learned about curl to sends commands instat off my browser but i still have some issues with the commands maby its worong format or somthing i just adding my commands and answer here and som log too.

i'm stuck and need the key :) ;) 

[log and adwseres.txt](https://github.com/johanmeijer/grott/files/10249430/log.and.adwseres.txt)

**Reply by chirs-51** [reply]
and here is to see the sever is on the ip and port 

C:\WINDOWS\system32>curl http://192.168.1.192:5782
<h2>Welcome to Grott the growatt inverter monitor</h2><br><h3>Made by Ledidobe, Johan Meijer</h3>

**Comment by johanmeijer** [comment]
Hi Chris, Seems al oke to me. What do you see in a browser when you use the following url's: 

http://192.168.1.192:5782/inverter
http://192.168.1.192:5782/datalogger

or 

curl http://192.168.0.206:5782/datalogger 

This normally gives generic information about your configuration. 

In my case (2 inverters, 2 dataloggers): 

    {"JPCxxxxxxx": {"ip": "192.168.0.206", "port": 32874, "protocol": "06", "QMBxxxxxxx": {"inverterno": "01", "power": 0}}, "JPCxxxxxxx": {"ip": "192.168.0.206", "port": 32888, "protocol": "06", "QNBxxxxxxxxx": {"inverterno": "01", "power": 0}}}

**Reply by chirs-51** [reply]
merry Christmas!!! hope you and your family gets a fantastic day

iam normally using chrome but i use curl ind cmd ind windows 10 

C:\Users\chris>curl http://192.168.1.192:5782/inverter
{"JPC6xxxxxx": {"ip": "127.0.0.1", "port": 48702, "protocol": "06", "OMCxxxxxxx": {"inverterno": "01", "power": 0}}}

C:\Users\chris>curl http://192.168.1.192:5782/datalogger
{"JPC6xxxxxx": {"ip": "127.0.0.1", "port": 48702, "protocol": "06", "OMCxxxxxxx": {"inverterno": "01", "power": 0}}}

just from cmd 

Chrome just has long waiting. 
http://192.168.1.192:5782/datalogger
{"JPCxxxxxxx": {"ip": "127.0.0.1", "port": 48702, "protocol": "06", "OMCxxxxxxx": {"inverterno": "01", "power": 0}}}


i can se the response ind the log too 
curl -X GET   192.168.1.192:5782/datalogger?command=register&register=4&datalogger=JPCxxxxxxx
log 192.168.1.215 - - [23/Dec/2022 15:48:35] "GET /datalogger?command=register HTTP/1.1" 400 -
	 - Grotthttpserver - http response send:  400 text/body b'no datalogger or inverterid specified'

i just switch my home assistant from running on a vm to just hassio on the hardware but it's seems to be wores now i dot get the right respose on 

C:\Users\chris>curl 192.168.1.192:5782/inverter?command=register&register=1091&inverter=OMCxxxxx
no or no valid invertid specified'register' is not recognized as an internal or external command,
operable program or batch file.
'inverter' is not recognized as an internal or external command,
operable program or batch file.

log 

192.168.1.215 - - [23/Dec/2022 16:01:12] "GET /inverter?command=register HTTP/1.1" 400 -
	 - Grotthttpserver - http response send:  400 text/html b'no or no valid invertid specified'
	 - Growatt packet received:
		  <socket.socket fd=4, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=0, laddr=('127.0.0.1',

can it be becurece my network is on 192.168.1.192 and not 192.168.0.192 ? i can't even check dot because of my internet provider


konfig file 
gmode: proxy
gnomqtt: false
gmqttip: 192.168.1.192
gmqttport: "1883"
gmqttauth: true
gmqtttopic: energy/grott
gmqttuser: xxxxxx
gmqttpassword: xxxxxx
ginvtype: sph
ggrowattip: 127.0.0.1
gverbose: true
ggrowattport: 5781
gincludeall: true
gsendbuf: false

**Reply by chirs-51** [reply]
![23 12 2022_21 48 06_REC](https://user-images.githubusercontent.com/72761662/209404293-a2c8f817-d464-4d03-92ed-31759a88be05.png)

iam pritty sure the port part its set here or do i need to open them inside my router?

**Comment by johanmeijer** [comment]
It just seems to be a Curl issue. I am not a curl expert but parsing the command does not seems to go wel. 

can you try in the browser.  

192.168.1.192:5782/datalogger?command=register&register=4&datalogger=JPCxxxxxxx

**Reply by chirs-51** [reply]
i switch to curl because of the trouble with chrome 

192.168.1.192:5782/datalogger?command=register&register=4&datalogger=JPCxxxxxxx
{"value": "5"}

the biggest issue is to get the commands to Work like 
http://192.168.1.192:5782/datalogger?command=register&register=4&datalogger=JPCxxxxx&value=0.1
i just get nothing or {"value": "5"} witch is wrong :/ when iam trying to make it 0.1 og even 1 
when i send the command with chrome the log just saying 

and log says 
192.168.1.215 - - [25/Dec/2022 22:03:00] "GET /datalogger?command=register&register=4&datalogger=JPCxxxxx&value=0.1 HTTP/1.1" 400 -
	 - Grotthttpserver - Get received 
192.168.1.215 - - [25/Dec/2022 22:03:00] "GET /favicon.ico HTTP/1.1" 200 -
	 - Grotthttpserver - http response send:  200 text/html b'<h2>Welcome to Grott the growatt inverter monitor</h2><br><h3>Made by Ledidobe, Johan Meijer</h3>'

i just feel iam so close to make it all working and just missing something.
